### PR TITLE
29 sat scripts should have as arguments filelists instead of dirs

### PR DESCRIPTION
**Pull request by gbolzon**
Refactoring
src/bitsea/Sat/sat_check_QI.py
src/bitsea/Sat/aveSat.py
src/bitsea/Sat/interpolator.py

in order to expose functions accepting and returning lists of filenames

### CODE REVIEW

File: src/bitsea/Sat/interpolator.py [hunk 1]
```diff
@@ -103,18 +106,19 @@ def argument():
...
 def interpolator(*,
 inmesh : str,
 serial : bool,
-maskfile : str,
-inputdir: str,
+maskfile : Path,
+inputfiles: List[str],
 outputdir:str,
 varnames: Iterable[str],
 force = bool,
```
Comment (spiani) [inline src/bitsea/Sat/interpolator.py:118]: This probably should be `force: bool`

File: src/bitsea/Sat/interpolator.py [hunk 2]
```diff
@@ -103,18 +106,19 @@ def argument():
...
 from bitsea.utilities.mpi_serial_interface import get_mpi_communicator
+from sys import exit
 
 def interpolator(*,
 inmesh : str,
 serial : bool,
-maskfile : str,
-inputdir: str,
+maskfile : Path,
+inputfiles: List[str],
```
Comment (spiani) [inline src/bitsea/Sat/interpolator.py:115]: `inputfiles` is a list of strings or a lists of `Paths`?

File: src/bitsea/Sat/aveSat.py [hunk 3]
```diff
@@ -58,27 +58,33 @@ def argument():
...
 from bitsea.utilities.mpi_serial_interface import get_mpi_communicator
 
 
 def aveSat(*,
-inputdir: Path,
+inputfiles: Iterable[Path],
+datetimes: List[str],
 outputdir: Path,
 mesh: str,
 varnames = List[str],
```
Comment (spiani) [inline src/bitsea/Sat/aveSat.py:78]: I think that this should be `varnames : List[str]`

**Review by spiani — APPROVED** [review]
@gbolzon , I already fixed the small typos I have pointed out in the messages. I also run `pre.commit` on all the files. Have a look and then, if you like it, we accept this pull request

File: src/bitsea/Sat/interpolator.py (hunk 1 above)
Comment (gbolzon) [reply src/bitsea/Sat/interpolator.py]: I agree

File: src/bitsea/Sat/interpolator.py (hunk 2 above)
Comment (gbolzon) [reply src/bitsea/Sat/interpolator.py]: I agree

File: src/bitsea/Sat/aveSat.py (hunk 3 above)
Comment (gbolzon) [reply src/bitsea/Sat/aveSat.py]: (y)

**Pull request merged**